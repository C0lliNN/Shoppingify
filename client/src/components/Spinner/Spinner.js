import React from 'react';
import styles from './Spinner.module.css';

export default function Spinner() {
  return (
    <div className={styles.Spinner}>
      <div className={[styles.Cube, styles.Cube1].join(' ')}></div>
      <div className={[styles.Cube, styles.Cube2].join(' ')}></div>
      <div className={[styles.Cube, styles.Cube3].join(' ')}></div>
      <div className={[styles.Cube, styles.Cube4].join(' ')}></div>
    </div>
  );
}
