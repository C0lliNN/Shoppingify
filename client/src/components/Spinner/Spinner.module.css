.Spinner {
  margin: 20px auto;
  width: 40px;
  height: 40px;
  position: relative;
  -webkit-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
}

.Spinner .Cube {
  float: left;
  width: 50%;
  height: 50%;
  position: relative;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}
.Spinner .Cube:before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgb(14, 156, 90);
  -webkit-animation: sk-foldCubeAngle 2.4s infinite linear both;
  animation: sk-foldCubeAngle 2.4s infinite linear both;
  -webkit-transform-origin: 100% 100%;
  -ms-transform-origin: 100% 100%;
  transform-origin: 100% 100%;
}
.Spinner .Cube2 {
  -webkit-transform: scale(1.1) rotateZ(90deg);
  transform: scale(1.1) rotateZ(90deg);
}
.Spinner .Cube3 {
  -webkit-transform: scale(1.1) rotateZ(180deg);
  transform: scale(1.1) rotateZ(180deg);
}
.Spinner .Cube4 {
  -webkit-transform: scale(1.1) rotateZ(270deg);
  transform: scale(1.1) rotateZ(270deg);
}
.Spinner .Cube2:before {
  -webkit-animation-delay: 0.3s;
  animation-delay: 0.3s;
}
.Spinner .Cube3:before {
  -webkit-animation-delay: 0.6s;
  animation-delay: 0.6s;
}
.Spinner .Cube4:before {
  -webkit-animation-delay: 0.9s;
  animation-delay: 0.9s;
}
@-webkit-keyframes sk-foldCubeAngle {
  0%,
  10% {
    -webkit-transform: perspective(140px) rotateX(-180deg);
    transform: perspective(140px) rotateX(-180deg);
    opacity: 0;
  }
  25%,
  75% {
    -webkit-transform: perspective(140px) rotateX(0deg);
    transform: perspective(140px) rotateX(0deg);
    opacity: 1;
  }
  90%,
  100% {
    -webkit-transform: perspective(140px) rotateY(180deg);
    transform: perspective(140px) rotateY(180deg);
    opacity: 0;
  }
}

@keyframes sk-foldCubeAngle {
  0%,
  10% {
    -webkit-transform: perspective(140px) rotateX(-180deg);
    transform: perspective(140px) rotateX(-180deg);
    opacity: 0;
  }
  25%,
  75% {
    -webkit-transform: perspective(140px) rotateX(0deg);
    transform: perspective(140px) rotateX(0deg);
    opacity: 1;
  }
  90%,
  100% {
    -webkit-transform: perspective(140px) rotateY(180deg);
    transform: perspective(140px) rotateY(180deg);
    opacity: 0;
  }
}
