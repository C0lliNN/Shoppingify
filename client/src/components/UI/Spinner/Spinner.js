import styled, { keyframes } from 'styled-components';
import { COLORS } from '../../../helpers/style-constants';

const rotate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

const Spinner = styled.div`
  border-radius: 50%;
  width: 10em;
  height: 10em;
  margin: 60px auto;
  font-size: 10px;
  position: relative;
  text-indent: -9999em;
  border-top: 1.1em solid rgba(249, 161, 9, 0.2);
  border-right: 1.1em solid rgba(249, 161, 9, 0.2);
  border-bottom: 1.1em solid rgba(249, 161, 9, 0.2);
  border-left: 1.1em solid ${COLORS.primary};
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-animation: ${rotate} 1.1s infinite linear;
  animation: ${rotate} 1.1s infinite linear;
  &:after {
    border-radius: 50%;
    width: 10em;
    height: 10em;
  }
`;

export default Spinner;
