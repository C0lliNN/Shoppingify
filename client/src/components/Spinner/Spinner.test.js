import Spinner from './Spinner';
import React from 'react';
import renderer from 'react-test-renderer';

test('A Spinner should have Spinner class', () => {
  const component = renderer.create(<Spinner />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
