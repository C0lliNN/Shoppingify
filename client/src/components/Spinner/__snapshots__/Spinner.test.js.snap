// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Spinner should have Spinner class 1`] = `
<div
  className="Spinner"
>
  <div
    className="Cube Cube1"
  />
  <div
    className="Cube Cube2"
  />
  <div
    className="Cube Cube3"
  />
  <div
    className="Cube Cube4"
  />
</div>
`;
